src/input_common/sdl/sdl_impl.h

```diff
@@ -9,6 +9,17 @@
 #include <mutex>
 #include <thread>
 #include <unordered_map>
+
+// Ignore -Wimplicit-fallthrough due to https://github.com/libsdl-org/SDL/issues/4307
+#ifdef __clang__
```

Comment:
```suggestion
#ifdef __GNUC__
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wimplicit-fallthrouh"
#endif
```

Will make this apply to both GCC and clang, since clang supports the GCC qualifier in pragmas

Ditto for the closing ifdef